validation, log: improve logging of ChainstateManager snapshot persistance

jonatack opened the pull request:
Use the `LOG_TIME_MILLIS_WITH_CATEGORY_MSG_ONCE` macro to improve the logging of ChainstateManager snapshot persistance, log task start and completion separately and no longer manually track the duration, as suggested by Marco Falke in https://github.com/bitcoin/bitcoin/pull/22872#discussion_r715571280.

Extract the flushing into one function, which clarifies the logic, extends the improved logging to both flushing call sites, and allows logging the prefix `FlushSnapshotToDisk`, which is similar to `FlushStateToDisk`.

before
```
[snapshot] flushing coins cache (0 MB)... done (0.00ms)

[snapshot] flushing snapshot chainstate to disk
```
after
```
FlushSnapshotToDisk: flushing coins cache (0 MB) started
...
FlushSnapshotToDisk: completed (0.00ms)

FlushSnapshotToDisk: saving snapshot chainstate (0 MB) started
...
FlushSnapshotToDisk: completed (0.00ms)
```
The logging can be observed in the output of
```
./src/test/test_bitcoin -t validation_chainstate_tests -- DEBUG_LOG_OUT
```

jonatack committed:
validation, log: extract FlushSnapshotToDisk() function

This moves the flushing and logging into one method and adds logging
of time duration and memory for the snapshot chainstate flushing.

jonatack committed:
validation, log: improve logging in FlushSnapshotToDisk()

Use the `LOG_TIME_MILLIS_WITH_CATEGORY_MSG_ONCE` macro to improve the
logging of snapshot persistance and no longer manually track the duration.

before

[snapshot] flushing coins cache (0 MB)... done (0.00ms)

[snapshot] flushing snapshot chainstate to disk (0 MB)... done (0.00ms)

after

FlushSnapshotToDisk: flushing coins cache (0 MB) started
FlushSnapshotToDisk: completed (0.00ms)

FlushSnapshotToDisk: saving snapshot chainstate (0 MB) started
FlushSnapshotToDisk: completed (0.00ms)

The logging can be observed in the output of

./src/test/test_bitcoin -t validation_chainstate_tests -- DEBUG_LOG_OUT

MarcoFalke commented on src/validation.cpp:
```diff
@@ -4840,6 +4840,20 @@ bool ChainstateManager::ActivateSnapshot(
+
+    const int64_t flush_now{GetTimeMillis()};
+
+    // TODO: if #17487 is merged, add erase=false here if snapshot is loaded, for better performance.
```
unrelated: Wouldn't it be better to remove this comment and just do this in `17487`?

jonatack replied:
Yes, done (I removed it in the first push here and then pushed again without it). Made a note in that PR: https://github.com/bitcoin/bitcoin/pull/17487#discussion_r767146237.

jonatack committed:
validation, doc: remove TODO comment

It would make for sense for the TODO to be done in PR 17487
(or noted in the review feedback for a follow-up),
no need to continue maintaining the TODO in the codebase.

jamesob commented:
Code review ACK https://github.com/bitcoin/bitcoin/pull/23738/commits/8e37fa83934ad47a4a73c326213e2073dfc35e29
